### PR TITLE
[Package] Backup: modify review request logic

### PR DESCRIPTION
**Pull request by nunyvega**
#### Changes proposed in this Pull Request:

Currently, we ask for a Jetpack Backup plugin review in the WordPress.org plugin repo whenever:
- the site completes a restore
- the site has a 3+ months active subscription.

Proposals on this PR: 
- Remove the 3+ months active subscription review request 
- Add a new review request whenever a site has 5+ successful backups in a row.  ( latest 5 backups )

#### Jetpack product discussion

P2 discussion: pbNhbs-3Il
Asana task: 1199562518309922-as-1202898039009401

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
- Create a site or use a preexisting site with a Jetpack Backup plan
- Optional but helpful: open the Rewind Debugger for the site to start backups if needed and see the 5 latest backups
- If you don't have 5 successful backups, confirm that the review request is not visible. If you do have them
- Complete 5 successful backups and confirm that the review request is visible
- Click on the review link and confirm that it works
- Dismiss the review request ( click on Maybe later ) and confirm that the review request disappears
- Reload the page and confirm that the message is not displayed anymore

#### Expected component:
Screen Shot 2022-09-01 at 14 10 20

### CODE REVIEW

File: projects/packages/backup/src/js/components/Admin.js [hunk 1]
```diff
@@ -205,29 +201,20 @@ const ReviewMessage = connectionLoaded => {
 		return false;
 	};
 
-	// Check if the site has an active Backup subscription older than 3 months
-	const hasThreeMonthsSub = () => {
-		if ( ! Array.isArray( purchases ) || purchases.length === 0 ) {
+	// Check if the last 5 backups were successful
+	const hasFiveSuccessfulBackups = () => {
+		if ( backups.length < 5 || ! Array.isArray( backups ) ) {
```
Comment (jwebbdev) [inline projects/packages/backup/src/js/components/Admin.js]: Should this check be reversed so you don't check the length of a non-array?

Comment (nunyvega) [reply projects/packages/backup/src/js/components/Admin.js]: Good point, I just corrected it

**Review by jwebbdev — APPROVED** [review]
Tested and worked on my jurassic.tube site. 👍

**Comment by nunyvega** [comment]
@Automattic/jetpack-crew I could use your help for a confidence check here:
Can you please confirm that this number of changelogs is expected? this change adds a modification to the connection package, which I guess explains it.

**Comment by anomiex** [comment]
You could probably delete `projects/packages/backup/changelog/modify-backup-review-request-logic#2` if you want, although it doesn't hurt anything.

Unfortunately, yeah, doing a minor bump of a package like connection means that every package that depends on it needs its dependency bumped too, which makes a lot of noise.

**Comment by nunyvega** [comment]
> You could probably delete `projects/packages/backup/changelog/modify-backup-review-request-logic#2` if you want, although it doesn't hurt anything.
> 
> Unfortunately, yeah, doing a minor bump of a package like connection means that every package that depends on it needs its dependency bumped too, which makes a lot of noise.

Thanks!
I  just wanted to be sure that it was expected :)
I'll enable auto-merge then.

**Review by sdixon194 — APPROVED** [review]

**Pull request merged**